components/dashboards-web-component/src/viewer/DashboardViewPage.jsx

```diff
@@ -272,6 +418,234 @@ class DashboardViewPage extends Component {
+                    pdf.addImage(imgData, 'JPEG', 0, 55, imgWidth * valH, imgHeight * valH, "dashboard" + ind, "FAST");
+                }
+
+            } else {
+                const val = (pdf.internal.pageSize.getHeight()) / imgHeight;
+                pdf = new jspdf("p", "px", "a4");
+                pdf.addImage(imgData, 'JPEG', 0, 35, imgWidth * val, imgHeight * val, "dashboard" + ind, "FAST");
+            }
+
+            // FOOTER : a customized footer defined by the user in the deployment.yaml file or else no footer by default
+            const str = "Page " + (ind + 1) + " of " + this.state.pageList.length;
+            pdf.setFontSize(10);
+            const pageHeight = pdf.internal.pageSize.height || pdf.internal.pageSize.getHeight();
+            pdf.text(str, 10, pageHeight - 10);
+
+            const footerImg = localStorage.getItem('dashboardFooter');
+            if (footerImg != null) {
+                pdf.addImage(footerImg, "JPEG", 380, 780, pdfConfig.stampImage.size.x, pdfConfig.stampImage.size.y);
+            }
+
+            if (ind < this.state.pageList.length - 1) {
+                pdf.addPage();
+
+                // HEADER : a cutomized header defined by the user or else the default stream processor header
+                const headerImg = localStorage.getItem('dashboardHeader');
+                if (headerImg != null) {
+                    pdf.addImage(headerImg, "JPEG", pdfConfig.stampImageDashboard.coordinates.x,
+                                 pdfConfig.stampImageDashboard.coordinates.y, pdfConfig.stampImageDashboard.size.x,
+                                 pdfConfig.stampImageDashboard.size.y);
+                }
+            }
+        });
+
+        clearInterval(this.timer);
+        this.handleReportStatusClose();
+        this.state.completed = 0;
+        this.state.includeTime = false;
+
+        const docName = title + ".pdf";
+        pdf.save(docName);
+
+    }
+
+    // Get the image from deployment.yaml file through the server
+    async addPdfImage(url, callback) {
+        let path = "/portal/public/app/images/"
+
+        await axios.get(url, { auth: { username: "admin", password: "admin" } })
+            .then(function (res) {
+                if (res.data === '') {
+                    callback('none');
+                } else {
+                    path += res.data;
+
+                    //Convert image to bdase64 encoded image (data_url)
+                    const canvas = document.createElement('canvas');
+                    const ctx = canvas.getContext('2d');
+                    const img = new Image();
+                    let imgStr;
+
+                    img.onload = function () {
+                        canvas.width = img.width;
+                        canvas.height = img.height;
+                        ctx.drawImage(img, 0, 0);
+                        imgStr = canvas.toDataURL("image/png");
```

Comment:
Use single quotes and double quotes as per the conventions. Please change other places as well.

Reply:
Done